src/transform.js

```diff
@@ -140,12 +140,57 @@ function transformShouldBeEqualToSpecific() {
   };
 }
 
+// Returns a closure that modifies the parameter, transformInfo
+// so that information gathered during traversal can be used in
+// the outer scope.
+function removeDescription(transformInfo) {
+  return function() {
+    return {
+      visitor: {
+        CallExpression(path) {
+          if (path.node.callee.name === 'description') {
+            // This will currently only take the first description.
+            // We might want to change to use the last, longest, or a
+            // concatenation?
+            if (!transformInfo.description) {
+              transformInfo.description = path.node.arguments[0].value;
+            }
+            path.remove();
+          }
+        },
+      },
+    };
+  };
+}
+
+// TODO: maybe change debug() to console.log() instead of deleting?
+function removeDebug() {
+  return {
+    visitor: {
+      CallExpression(path) {
+        if (path.node.callee.name === 'debug') {
+          path.remove();
+        }
+      },
+    },
+  };
+}
+
+// Applies transformations in pluginArray to sourceCode (string)
+// If addSetup is true, will add setup() test call at the beginning of
+// the script.
+// Returns an object with the transformed source code and the test title,
+// if one was parsed from description() calls.
 function transformSourceCodeString(sourceCode, addSetup=true) {
+  const transformInfo = {};
```

Comment:
can you explain somewhere what transformInfo's structure is intended to be?

Reply:
Sure, I'll add a comment here. It's basically a hack to pass a mutable object to the plugin returned closure so we can get data from inside the transformation code back to the main function scope.